detox/android/detox/src/main/java/com/wix/detox/espresso/EspressoDetox.java

```diff
@@ -34,10 +34,13 @@
 public class EspressoDetox {
     private static final String LOG_TAG = "detox";
 
-    private static final String METHOD_GET_ACTIVITY = "getCurrentActivity";
+    public static Object perform(ViewInteraction interaction, ViewAction action) {
+        interaction.perform(action);
 
-    public static ViewInteraction perform(ViewInteraction interaction, ViewAction action) {
-        return interaction.perform(action);
```

Comment:
This never made sense! `interaction.peform()` always returns the interaction itself. `invoke` cannot parse an object, and therefore the returned result has _always_ been `null`.

With this new implementation, it's possible to return a result from the `action` itself. In the future, this should be useful for implementing `getAttributes()` @rotemmiz